4/README.md

```diff
@@ -0,0 +1,346 @@
+* *period*, *archive period*, describing the minimum time period
+  in milliseconds after which a *PERIODIC* or *ARCHIVE* event MUST be reported,
+  regardless of Attribute's *read value*. If *period* is not specified,
+  a default value of 1000 ms is used,
+  It MUST be defined only for Attributes with numerical *data type*.
+
+See [Attribute events section](#attribute-events) for more details.
+
+```abnf
+rel-change = change
+abs-change = chagne
+
+archive-rel-change = change
+archive-abs-change = chagne
+
+number = [ "-" ] 1*DIGIT [ "." ] *DIGIT
+change = number [ "," number ]
+
+period = [ "-" ] 1*DIGIT
+archive-period = period
+```
+
+### Attribute runtime parameters
+
+At given point in time an Attribute MUST have associated:
+* *quality*, an enumeration describing the state *read value*
+  (one of *VALID*, *INVALID*, *ALARM*, *CHANGING*, *WARNING*),
+* *read value*, an object representing the value of the Attribute.
+  It MUST conform to *data format* and *data type*,
+* *read dim x*, an integer describing the number of data elements in
+  *read value* in X dimension.
+  If *data format* is *SCALAR*, it MUST be either 0 or 1.
+  If *data format* is *SPECTRUM* or *IMAGE*, it MUST be between 0 and
+  *max dim x*.
+* *read dim y*, an integer describing the number of data elements in
+  *read value* in Y dimension.
+  If *data format* is *SCALAR* or *SPECTRUM*, it MUST be 0.
+  If *data format* is *IMAGE*, it MUST be between 0 and *max dim y*.
+
+Additionally, if *writable* is *WRITE* or *READ_WRITE*,
+at given point in time an Attribute MUST have associated:
+* *set value*, an object representing the value set to the Attribute.
+  It MUST conform to *data format* and *data type*,
+* *write dim x*, an integer describing the number of data elements in
+  *set value* in X dimension,
+* *write dim y*, an integer describing the number of data elements in
+  *set value* in Y dimension.
+
+
+### Attribute aliases
+
+A *full attribute name* consists of *device name* and *attribute name*
+separated by "/" character.
+
+See [RFC 2/Device](/2) for the definition of `<device-name>`.
+
+An Attribute MAY have associated *alias*, a string which can be used
+in place of *full attribute name* to address the Attribute.
+An Attribute MUST NOT have more than one *alias*.
+
+An *alias* MUST be stored in the database.
+
+An *alias* MUST be unique.
+
+An *alias* MAY contain any character (including multi-byte characters) except
+that it MUST NOT contain any of `/`, ` ` (space), `#`, `:`, `->`.
+It MAY be empty.
+See `<attribute-alias>` specification below.
+
+
+### Attribute events
+
+An Attribute can send following events:
+* *PERIODIC*,
+* *CHANGE*,
+* *ARCHIVE*,
+* *PIPE*,
+* *DATA READY*,
+* *ATTR CONF*,
+* *USER*,
+
+In order to send *PERIODIC*, *CHANGE*, *ARCHIVE*, *PIPE* and *DATA READY*
+events, an attribute must be polled. See
+[Attribute polling section](#attribute-events) for more details.
```

Comment:
This is not true. Events can be pushed by code. The polling is not mandatory in this case.

Reply:
> Proposed fix: replace above text with:
> 
> Events can be send automatically by a polling mechanism, automaticallyby the Tango system or manually from the device server code.

can be send => can be sent
automaticallyby  => automatically by 

> 
> The polling mechanism can send _PERIODIC_, _CHANGE_ and _ARCHIVE_ events.
> In order to send events via the polling mechanism, the polling for the Attribute MUST be configured in the Device with _polled attr_ Property.

I would just write something like:
"In order to send events via the polling mechanism, the polling for the Attribute MUST be enabled"
without going into more details. The polling RFC can give all the details on how to configure/start the polling for a given attribute.

> The device server code can manually send _CHANGE_, _ARCHIVE_, _DATA READY_ and _USER_ events without the need to configure polling for the Attribute.

I think we should remove the mention to PIPE events a bit above too. PIPE are not attributes so we should not mention that in this RFC which is dedicated to attributes.